exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/diagnosiskey/DiagnosisKeyDao.java

```diff
@@ -118,6 +125,81 @@ public boolean verify(int verificationId, String requestChecksum) {
         }
     }
 
+    @Transactional(timeout = 10)
+    public long startOutboundOperation() {
+        String lock = "lock table en.efgs_operation in access exclusive mode";
+        jdbcTemplate.update(lock, Map.of());
+        if (isOutboundOperationAvailable()) {
```

Comment:
I don't understand. I think @Transactional timeout with jdbctemplate sets timeout at jdbc statement level before each statement, so if app is working this should be enough. Or even if it's shut-down without total crash. However, in case of crash in the middle of transaction it's probably more complicated thing. Then it's most likely about when the underlying tcp connection to db get killed and after that db will do rollback and releases the lock on the same time. Problematic situation would be the case where connection stays alive even the app has been crashed on the right moment. We might send statement_timeout before actual statement to db to be sure, but let's think this through with patience.

Reply:
What the...? This comment was written ages ago, and on a very different version of this file. It must have been left-over from some uncommitted review and pulled into this one. I didn't write it today. Just ignore it, I'll check the retry-logic from the current code and comment if needed.

Reply:
I checked how the mid-operation failures work in the current implementation and I don't see any issue with that.